apps/ledger-live-mobile/src/newArch/features/FirmwareUpdate/components/UpdateBanner/UpdateBanner.test.tsx

```diff
@@ -0,0 +1,399 @@
+            wired: false, // Device is not wired
+          }),
+        },
+        appstate: {
+          ...state.appstate,
+          hasConnectedDevice: true,
+        },
+      }),
+    });
+
+    // Check that the banner is displayed with the correct wording
+    expect(await screen.findByText("OS update available")).toBeOnTheScreen();
+    expect(
+      await screen.findByText("Tap to update your Ledger Nano X to OS version mockVersion."),
+    ).toBeOnTheScreen();
+
+    // Press the banner and check that the unsupported drawer is displayed
+    await user.press(screen.getByTestId("fw-update-banner"));
+    expect(navigateToOldUpdateFlow).not.toHaveBeenCalled();
+    expect(navigateToNewUpdateFlow).not.toHaveBeenCalled();
+    expect(await screen.findByText("USB cable needed")).toBeOnTheScreen();
+    expect(
+      await screen.findByText(
+        "To start the firmware update, plug your Ledger Nano X to your mobile phone using a USB cable.",
+      ),
+    ).toBeOnTheScreen();
+  });
+
+  oldUpdateFlowNotSupportedDataSet.forEach(({ deviceModelId, version, productName }) => {
+    it(`should open the unsupported drawer if there is an update and it's Android but the update is not supported for this device version (${version} ${deviceModelId})`, async () => {
+      PlatformSpy.mockReturnValue({ OS: "android" } as typeof ReactNative.Platform);
+      useLatestFirmware.mockReturnValue({
+        final: {
+          name: "mockVersion",
+        },
+      });
+
+      const { user } = render(<UpdateBanner onBackFromUpdate={() => {}} />, {
+        overrideInitialState: (state: State) => ({
+          ...state,
+          settings: {
+            ...state.settings,
+            ...makeMockSettings({
+              modelId: deviceModelId,
+              version,
+              hasCompletedOnboarding: true,
+              wired: true, // Device is wired
+            }),
+          },
+          appstate: {
+            ...state.appstate,
+            hasConnectedDevice: true,
+          },
+        }),
+      });
+
+      // Check that the banner is displayed with the correct wording
+      expect(await screen.findByText("OS update available")).toBeOnTheScreen();
+      expect(
+        await screen.findByText(
+          `Tap to update your Ledger ${productName} to OS version mockVersion.`,
+        ),
+      ).toBeOnTheScreen();
+
+      // Press the banner and check that it doesn't navigate to the update flow
+      await user.press(screen.getByTestId("fw-update-banner"));
+      expect(navigateToOldUpdateFlow).not.toHaveBeenCalled();
+      expect(navigateToNewUpdateFlow).not.toHaveBeenCalled();
+
+      // Check that the unsupported drawer is displayed
+      expect(await screen.findByText("Firmware Update")).toBeOnTheScreen();
+      expect(
+        await screen.findByText(
+          "Update your Ledger Nano firmware by connecting it to the Ledger Live application on desktop",
+        ),
+      ).toBeOnTheScreen();
```

Comment:
You should always assert the change in the UI before anything else. Because if you don't assert that the change is done you can assert wrongly that something has not been called by being too quick in execution.
For example, here we could have:
`navigateToOldUpdateFlow` or `navigateToNewUpdateFlow` has been called after the change in the UI but as you assert it just after the action without checking a change in the UI we are not sure if it works thanks to execution time or because it really works.